tests/test_redirects.py

```diff
@@ -0,0 +1,156 @@
+import json
+
+from test_grapple import BaseGrappleTest
+from testapp.factories import RedirectFactory
+from testapp.models import BlogPage
+from wagtail_factories import SiteFactory
+
+
+class TestRedirectQueries(BaseGrappleTest):
+    def setUp(self) -> None:
```

Comment:
**Question:** Why `setUp` not `setUpTestData`? This looks like data that can be shared between all test methods within this class.

Reply:
I believe it still is shared between all methods. I use the definitions from `setUp` in `test_new_url_sources`

Reply:
`setUp` runs for each test within a test case, whereas `setUpTestData` is run once for the class and is available for all methods to use.

So while it seems like `setUp` is shared between methods within a class, it's actually redoing the work `n` times.

Reply:
Thanks, addressed in 33561505891fe6ec7d6a5901f9301af1d26aaad5